docs/sequence/PISPTransactionApi.puml

```diff
@@ -27,125 +41,317 @@ rnote right of PISP #Light
 {
   signedChallenge: "base64(xxx)"
 }
 end note
-PISP -> OBS: ""POST /thirdpartyTransactions/{id}/approve""
+PISP -> PISP_TP_OUT: POST /thirdpartyTransactions/{transactionRequestId}/approve
 activate PISP
-activate OBS
+activate PISP_TP_OUT
+
+PISP_TP_OUT -> PISP_TM: approve()
+activate PISP_TM
+rnote right of PISP_TM: state: approvalReceived
+PISP_TM -> PISP_TM: SDKRequest.verifyAuthorization(transactionRequestId, signedChallenge)
+
+PISP_TM -> Switch: PUT /authorizations/{transactionRequestId}
+activate Switch
+Switch --> PISP_TM: 202 Accepted
+rnote right of PISP_TM: state: pendingVerifyAuthorization
+deactivate PISP_TM
+Switch -> DFSP_SDK_IN: PUT /authorization/{transactionRequestId}
+activate DFSP_SDK_IN
+DFSP_SDK_IN --> Switch: 202 Accept
+deactivate Switch
+DFSP_SDK_IN -> DFSP_SDK_OUT: notify about PUT /authorization/{transactionRequestId}
+activate DFSP_SDK_OUT
+DFSP_SDK_OUT -> DFSP_TM: 200: AuthorizationInformation
+note over DFSP_TM
+  From now DFSPTransactionModel is unblocked
+end note
+deactivate DFSP_SDK_IN
+deactivate DFSP_SDK_OUT
+note over DFSP_TM
+  DFSP checks the received AuthorizationInformation with backend.
+end note
+rnote right of DFSP_TM: state: pendingVerifyAuthorization
+DFSP_TM -> DFSP_TM: PISPBackendRequest.verifyAuthorization(AuthorizationInformation)
 
-OBS -> ST8: ""approve()""
-activate ST8
-rnote right of ST8: ""state: approvalReceived""
+note over PISP, DFSP_TM
+  ERROR_SCENARIO: DFSP notifies PISP about Authorization error and then ends its workflow here
+end note
+rnote right of DFSP_TM: DFSP is unhappy with Authorization\n**transactionRequestState** is **REJECTED**
+rnote right of DFSP_TM: state: failureVerifyAuthorization
+DFSP_TM -> DFSP_TM: persist transactionRequestState for transactionRequestId
+DFSP_TM -> DFSP_TM: SDKRequest.notifyThirdpartyAboutRejectedAuthorization(transactionRequestId)
+DFSP_TM -> Switch: PUT /thirdpartyRequests/transaction/{transactionRequestId}/error
+activate Switch
+Switch -> DFSP_TM: 202 Accept
+note over DFSP_TM
+  ERROR_SCENARIO: **DFSPTransactionModel ends its workflow here**
+end note
+Swich -> PISP_TP_IN: PUT /thirdpartyRequests/transaction/{transactionRequestId}/error
+activate PISP_TP_IN
+PISP_TP_IN -> Switch: 202 Accept
+deactivate Switch
+
+note over PISP_TM
+  ERROR_SCENARIO: **PISPTransactionModel will end its workflow here**
+end note
+PISP_TP_IN -> PISP_TM: authorizationRejected()
+deactivate PISP_TP_IN
+activate PISP_TM
+rnote right of PISP_TM: state: rejectedAuthorization
+PISP_TM -> PISP_TP_OUT: not approved: authorization is rejected
+deactivate PISP_TM
+PISP_TP_OUT -> PISP: 500: authorization is rejected
 
 
-ST8 -> Switch: ""PUT /authorizations/{id}""
-Switch --> ST8: ""202 Accepted""
-rnote right of ST8: ""state: pendingTransaction""
 
-note over Switch
-  1. DFSP checks the authorization with the auth-service, or their own auth-service.
-  2. DFSP is happy, so initiates the ""POST /transfer""
+note over PISP, DFSP_TM
```

Comment:
This is being cut off on the `.png` render - I think you need to give the puml renderer more memory (or switch to svg)

Reply:
replaced png by svg